app/src/main/java/org/piramalswasthya/sakhi/ui/home_activity/infant/InfantFormFragment.kt

```
@@ -0,0 +1,127 @@
+                        val isEditable = when (day) {
+                            nextEditable -> true
+                            "14th Day", "21st Day", "28th Day" -> "7th Day" in completed && day !in completed
+                            "42nd Day" -> "7th Day" in completed && "28th Day" in completed && day !in completed
+                            else -> false
+                        }
+
+                        val visitDate = visit?.let {
+                            try {
+                                val json = JSONObject(it.formDataJson)
+                                json.optString("visitDate", "-")
+                            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)